packages/renderer/src/lib/preferences/PreferencesConnectionCreationRendering.svelte

```diff
@@ -6,20 +6,59 @@ import type { Logger as LoggerType } from '@podman-desktop/api';
 import Logger from './Logger.svelte';
 import { writeToTerminal } from './Util';
 import ErrorMessage from '../ui/ErrorMessage.svelte';
+import { filesize } from 'filesize';
+import { onMount } from 'svelte';
+import { router } from 'tinro';
 export let properties: IConfigurationPropertyRecordedSchema[] = [];
 export let providerInfo: ProviderInfo;
 export let propertyScope: string;
 export let callback: (param: string, data, logger: LoggerType) => Promise<void>;
 
+$: configurationValues = new Map<string, string>(); 
 let creationInProgress = false;
 let creationStarted = false;
 let creationSuccessful = false;
+let pageIsLoading = true;
 
+let osMemory, osCpu, osFreeDisk;
 // get only ContainerProviderConnectionFactory scope fields that are starting by the provider id
-let configurationKeys: IConfigurationPropertyRecordedSchema[];
-$: configurationKeys = properties
+let configurationKeys: IConfigurationPropertyRecordedSchema[] = [];
+
+onMount(async () => {
+  osMemory = await window.getOsMemory();
+  osCpu = await window.getOsCpu();
+  osFreeDisk = await window.getOsFreeDiskSize();
+  configurationKeys = properties
   .filter(property => property.scope === propertyScope)
-  .filter(property => property.id.startsWith(providerInfo.id));
+  .filter(property => property.id.startsWith(providerInfo.id))
+  .map(property => {
+    switch (property.maximum) {
+      case 'HOST_TOTAL_DISKSIZE': {
+        if (osFreeDisk) {
+          property.maximum = osFreeDisk;  
+        }            
+        break;
```

Comment:
Currently i only show the free disk available when you try to create apodman machine, so if you have 20GB it will only allow you to allocate 20GB.

But as it calculate the actual free disk space, if you create 25 machine without starting them you will always be able to allocate those 20GB every time you create a new machine.

I see the problem you're raising but i don't know how to behave tbh. Any idea?